### PR TITLE
docs: move public folder to workspace configuration section

### PR DESCRIPTION
**Pull request by mvdluit**
move the public folder entry from the application project files section to the workspace configuration files section

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/main/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [x] Documentation content changes
- [ ] angular.io application / infrastructure changes
- [ ] Other... Please describe:

## What is the current behavior?

https://angular.dev/reference/configs/file-structure#application-source-files shows `public` as a sub-folder in `src`. This is incorrect, the `public` folder is created in the root folder.

Issue Number: N/A

## What is the new behavior?
Lists the `public` folder under https://angular.dev/reference/configs/file-structure#workspace-configuration-files to correctly reflect the file structure of a v18 project.

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: adev/src/content/reference/configs/file-structure.md [hunk 1]
```diff
@@ -34,14 +34,15 @@ All projects within a workspace share a [configuration](reference/configs/worksp
...
 
 | Workspace configuration files | Purpose                                                                                                                                                                                                                                                                                                          |
-|:---                           |:---                                                                                                                                                                                                                                                                                                              |
+| :---------------------------- | :--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
```
Comment (JeanMeche) [inline adev/src/content/reference/configs/file-structure.md:37]: Can you make sure to revert those table formating changes. We'd like to keep our docs consistent. 
Thank you.

Comment (mvdluit) [reply adev/src/content/reference/configs/file-structure.md]: Fixed.

**Review by JeanMeche — APPROVED** [review]
Thank you for this !

**Comment by atscott** [comment]
This PR was merged into the repository by commit 78c44ec5ec040d0a216ea3c59f76b0df8779cc58.

The changes were merged into the following branches: main, 18.1.x

**Pull request closed without merging**